otel-extension/src/main/scala/io/scalac/mesmer/otelextension/instrumentations/zio/ConcurrentMetricRegistryPoller.scala

```diff
@@ -0,0 +1,60 @@
+package io.scalac.mesmer.otelextension.instrumentations.zio
+
+import java.util.Timer
+import java.util.TimerTask
+
+import zio.metrics.MetricKey
+import zio.metrics.MetricKeyType
+
+import scala.collection.mutable
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+import io.scalac.mesmer.otelextension.instrumentations.zio.ZIOMetrics._
+
+class ConcurrentMetricRegistryPoller {
+
+  private val timer = new Timer()
+
+  private val instruments = mutable.HashMap.empty[MetricKey[MetricKeyType], AutoCloseable]
+
+  private def task = new TimerTask {
+    override def run(): Unit =
+      try {
+        val snapshot = ConcurrentMetricRegistryClient.snapshot()
+
+        snapshot.filter { metricPair =>
+          !instruments.contains(metricPair.metricKey)
+        }.foreach { metricPair =>
+          val autocloseable = metricPair.metricKey.keyType match {
+            case _: MetricKeyType.Counter =>
+              registerCounterAsyncMetric(metricPair.metricKey.asInstanceOf[MetricKey.Counter])
+            case _: MetricKeyType.Gauge =>
+              registerGaugeAsyncMetric(metricPair.metricKey.asInstanceOf[MetricKey.Gauge])
+            case _ =>
+              // TODO setup sync instruments for histograms
```

Comment:
Please create an issue for that

Reply:
https://github.com/ScalaConsultants/mesmer/issues/640

Reply:
Added to 0.9.0